app/services/topics/mutator.rb

```diff
@@ -71,6 +72,40 @@ def attach_files(signed_ids)
     end
   end
 
+  def rename_files(signed_ids)
+    return if signed_ids.blank?
+    signed_ids.each do |signed_id|
+    document = topic.documents.find { |doc| doc.blob.signed_id == signed_id }
+      next unless document
+
+      new_filename = custom_file_name(document)
+      rename_document(document, new_filename)
+    end
+  end
+
+  def rename_document(document, new_filename)
+    return unless document.blob.filename.to_s.split("_").first == "rename"
+    return if document.filename == new_filename
+
+    file_io = StringIO.new(document.download)
+      topic.documents.attach(
+        io: file_io,
+        filename: new_filename,
+        content_type: document.content_type
+      )
+      document.purge
+  end
+
+  def custom_file_name(document)
```

Comment:
Can  we move this method to topic? Looks like topic is mentioned here more frequently